Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -624,18 +633,22 @@ extension ActivityStreamPanel: DataObserverDelegate {
 
         let removeTopSiteAction = ActionOverlayTableViewAction(title: Strings.RemoveFromASContextMenuTitle, iconString: "action_close", handler: { action in
             self.telemetry.reportEvent(.Remove, source: pingSource, position: index)
-            self.hideURLFromTopSites(site.tileURL)
+            self.hideURLFromTopSites(site)
         })
 
         let dismissHighlightAction = ActionOverlayTableViewAction(title: Strings.RemoveFromASContextMenuTitle, iconString: "action_close", handler: { action in
             self.telemetry.reportEvent(.Dismiss, source: pingSource, position: index)
             self.hideFromHighlights(site)
         })
 
+        let pinTopSite = ActionOverlayTableViewAction(title: "Pin TopSite", iconString: "action_share", handler: { action in
```

Comment:
Oh and the title should change depending on if the site is pinned or not. If its already been pinned it should say "Unpin site" or something.